sdk/spring/azure-spring-boot/src/main/java/com/azure/spring/aad/AADTrustedIssuerRepository.java

```diff
@@ -22,30 +22,48 @@
 public class AADTrustedIssuerRepository {
 
     private static final Logger LOGGER = LoggerFactory.getLogger(AADTrustedIssuerRepository.class);
+
     private static final String LOGIN_MICROSOFT_ONLINE_ISSUER = "https://login.microsoftonline.com/";
+
     private static final String STS_WINDOWS_ISSUER = "https://sts.windows.net/";
+
     private static final String STS_CHINA_CLOUD_API_ISSUER = "https://sts.chinacloudapi.cn/";
+
     private static final String PATH_DELIMITER = "/";
+
     private static final String PATH_DELIMITER_V2 = "/v2.0";
-    private final List<String> trustedIssuers = new ArrayList<>();
+
     private final String tenantId;
 
+    /**
+     * In the trustedIssuersMap, the key stores the trusted Issuer, and the value stores the corresponding well known
+     * base uri.
+     */
+    private final Map<String, String> trustedIssuersMap = new HashMap<>();
+
     public AADTrustedIssuerRepository(String tenantId) {
         this.tenantId = tenantId;
-        trustedIssuers.addAll(buildAADIssuers(PATH_DELIMITER));
-        trustedIssuers.addAll(buildAADIssuers(PATH_DELIMITER_V2));
+        trustedIssuersMap.putAll(buildAADIssuers(PATH_DELIMITER));
+        trustedIssuersMap.putAll(buildAADIssuers(PATH_DELIMITER_V2));
     }
 
-    private List<String> buildAADIssuers(String delimiter) {
-        return Arrays.asList(LOGIN_MICROSOFT_ONLINE_ISSUER, STS_WINDOWS_ISSUER, STS_CHINA_CLOUD_API_ISSUER)
+    private Map<String, String> buildAADIssuers(String delimiter) {
+        return Arrays.asList(LOGIN_MICROSOFT_ONLINE_ISSUER, STS_WINDOWS_ISSUER,
+            STS_CHINA_CLOUD_API_ISSUER)
                      .stream()
                      .map(s -> s + tenantId + delimiter)
-                     .collect(Collectors.toList());
+                     .collect(Collectors.toMap(String::toString, String::toString));
     }
 
-    public void addB2CIssuer(String baseUri) {
-        Assert.notNull(baseUri, "tenantName cannot be null.");
-        trustedIssuers.add(String.format(resolveBaseUri(baseUri) + "/%s/v2.0/", tenantId));
+    public void addB2CUserFlowIssuers(String baseUri, Map<String, String> userFlows) {
+        Assert.notNull(baseUri, "baseUri cannot be null.");
+        Assert.notNull(userFlows, "userFlows cannot be null.");
+        userFlows.keySet().forEach(key -> createB2CIssuer(resolveBaseUri(baseUri), userFlows.get(key)));
+    }
+
+    private void createB2CIssuer(String resolveBaseUri, String userFlowName) {
```

Comment:
what if there are 2 userflows with the same baseuri and tenantId, will it have problem here?